src/lib/core/PeerId.h

```diff
@@ -39,15 +40,49 @@ class PeerId
         return *this;
     }
 
-    FabricId GetFabricId() const { return mFabricId; }
-    PeerId & SetFabricId(FabricId id)
+    CompressedFabricId GetCompressedFabricId() const { return mCompressedFabricId; }
+    PeerId & SetCompressedFabricId(CompressedFabricId id)
+    {
+        mCompressedFabricId = id;
+        return *this;
+    }
+
+    bool operator==(const PeerId & other) const
+    {
+        return (mNodeId == other.mNodeId) && (mCompressedFabricId == other.mCompressedFabricId);
+    }
```

Comment:
This is very likely true, but not exactly true. We have to be careful, because if this is used in ACL processing one day, it is a security issue. There needs to be a match against root public key as well. I think PeerId could be refactored to always refer to a local fabric index, and have getters that index into fabrics info metadata to recover either the root public key, fabric ID or compressed fabric ID. This way, you could do full check against root as well.

Reply:
Is the concern that there could be different combinations of `<uncompressed Fabric ID, Root pubkey>` that can yield the same compressed fabric ID?

Reply:
I tried making `PeerId` carry a `P256Pubkey` in it (for the root pub key). That's causing circular dependencies, unfortunately. So we'll need to move `PeerId.h` out of `src/lib/core` to some module that's already dependent on `src/Crypto`.

I'll submit an issue to track it.

Reply:
https://github.com/project-chip/connectedhomeip/issues/9192

Reply:
If `src/Crypto` depends on `PeerId`, there is an abstraction problem and we can resolve it.

The PeerID doesn't need to carry the key itself. It can carry FabricId, and that's used to later lookup a key if/when needed

And re:

> Is the concern that there could be different combinations of <uncompressed Fabric ID, Root pubkey> that can yield the same compressed fabric ID?

The answer is yes. For discovery, it's not a problem, since the worst case is a denial of service. For access control  enforcement of subject or CASE, the tuple <Root public key, uncompressed fabric id, node id> is what must explicitly be matched (somehow, not necessarily through the bits, but at least in a way that it's not spoofable/collidable).